seqspec/seqspec_check.py

```diff
@@ -453,20 +453,17 @@ def seqtype_check(rgn, errors, idx):
                     "error_message": f"'{rgn.region_id}' sequence_type is 'random' and contains subregions",
                     "error_object": "region",
                 }
-                # errors.append(
-                #     f"[error {idx}] '{rgn.region_id}' sequence_type is 'random' and contains subregions"
-                # )
                 errors.append(errobj)
                 idx += 1
-            if rgn.sequence_type == "random" and rgn.sequence != "X" * rgn.max_len:
+            if rgn.sequence_type == "random" and (
+                set(rgn.sequence) != {"X"}
+                or not (rgn.min_len <= len(rgn.sequence) <= rgn.max_len)
```

Comment:
Not sure if this makes sense, but seems to me like the sequence length not being between the min and max is an error in general and not only in this case?

Reply:
ok, this is checked elsewhere.